tool/run_leak_harness.sh

```
@@ -0,0 +1,167 @@
+  test 2>&1 | tee "$XCODE_LOG" | tail -50
+TEST_EXIT=${PIPESTATUS[0]}
+echo "[leak-harness] xcodebuild test exit=$TEST_EXIT"
+
+# ── 3. Stop server + scan ───────────────────────────────────────────────────
+kill -TERM "$LOG_STREAM_PID" 2>/dev/null || true
+LOG_STREAM_PID=""
+kill -TERM "$SERVER_PID" 2>/dev/null || true
+wait "$SERVER_PID" 2>/dev/null || true
+SERVER_PID=""
+
+FRAME_COUNT=$(ls "$DIR"/*.jpg 2>/dev/null | wc -l | tr -d ' ')
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)